CMSIS/NN/Source/NNSupportFunctions/arm_nn_mat_mul_core_4x_s8.c

```diff
@@ -57,40 +58,40 @@ arm_status arm_nn_mat_mul_core_4x_s8(const int32_t row_elements,
     int32_t acc_n3 = 0;
 
     const int8_t *ip_row_0 = row_base;
-    const int8_t *ip_row_1 = row_base + row_elements;
-    const int8_t *ip_row_2 = row_base + (2 * row_elements);
-    const int8_t *ip_row_3 = row_base + (3 * row_elements);
+    const int8_t *ip_row_1 = row_base + offset;
+    const int8_t *ip_row_2 = row_base + (2 * offset);
+    const int8_t *ip_row_3 = row_base + (3 * offset);
     int32_t sum_tmp = 0;
 
-#if defined(ARM_MATH_MVEI)
-        __asm volatile (
-           "   vldrb.8         q0, [%[col]], 16     \n"
-           "   wlstp.8         lr, %[cnt], 1f       \n"
-           "2:                                      \n"
-           "   vaddva.s8      %[sum], q0            \n"
-           "   vldrb.8         q1, [%[row0]], 16    \n"
-           "   vmladava.s8    %[out0], q0, q1       \n"
-           "   vldrb.8         q2, [%[row1]], 16    \n"
-           "   vmladava.s8     %[out1], q0, q2      \n"
-           "   vldrb.8         q3, [%[row2]], 16    \n"
-           "   vmladava.s8     %[out2], q0, q3      \n"
-           "   vldrb.8         q4, [%[row3]], 16    \n"
-           "   vmladava.s8     %[out3], q0, q4      \n"
-           "   vldrb.8         q0, [%[col]], 16     \n"
-           "   letp            lr, 2b               \n"
-           "1:                                      \n"
-           :[col] "+r"(col_base)
-           ,[sum] "+Te"(sum_tmp)
-           ,[row0] "+r"(ip_row_0)
-           ,[row1] "+r"(ip_row_1)
-           ,[row2] "+r"(ip_row_2)
-           ,[row3] "+r"(ip_row_3)
-           ,[out0] "+Te"(acc_n0)
-           ,[out1] "+Te"(acc_n1)
-           ,[out2] "+Te"(acc_n2)
-           ,[out3] "+Te"(acc_n3)
-           :[cnt] "r"(row_elements)
-           :"q0","q1", "q2","q3","q4", "memory", "r14");
+#if defined(ARM_MATH_MVEI) && !defined(ARM_MATH_AUTOVECTORIZE)
```

Comment:
I still think another flag is needed to select C or Asm (in case this Asm specific syntax is not supported on a given C compiler).
It is not the goal of ARM_MATH_AUTOVECTORIZE flag which can also be applied to C code.
ARM_MATH_AUTOVECTORIZE is used to select a version of the code which is working well with auto vectorizer (so no unrolling, not too many intrisinsics ...)

Reply:
@christophe0606 ok. i'll create a new one..